File: 36-valid-sudoku/valid-sudoku.py
class Solution(object):
    def isValidSudoku(self, board):
        for i in range(9):
            for j in range(9):
                if(board[i][j]=="."):
                    board[i][j]=10;
                else:
                    board[i][j]=int(board[i][j]);
        for  i in board:
            a=[0]*10;
            for j in i:
                if(j<10):
                    a[j]+=1;
                    if(a[j]>1):
                        return False;
        for i in range(len(board)):
            x=[];
            for j in range(len(board)):
                x.append(board[j][i]);
            a=[0]*10;
            for j in x:
                if(j<10):
                    a[j]+=1;
                    if(a[j]>1):
                        return False;
        for i in range(3):
            for j in range(3):
                x=[];
                for r in range(3*i,(i+1)*3):
                    for c in range(3*j,(j+1)*3):
                        x.append(board[r][c]);
                a=[0]*10;
                for ele in x:
                    if(ele<10):
                        a[ele]+=1;
                        if(a[ele]>1):
                            return False;             
        return True;
        """
        :type board: List[List[str]]
        :rtype: bool
        """
        

File: 36-valid-sudoku/valid-sudoku.java
class Solution {
    public boolean isValidSudoku(char[][] board) {
        List<Set<Character>> rows=new ArrayList<>();
        List<Set<Character>> cols=new ArrayList<>();
        List<Set<Character>> boxes=new ArrayList<>();
        for(int i=0;i<9;i++){
            rows.add(new HashSet<>());
            cols.add(new HashSet<>());
            boxes.add(new HashSet<>());
        }
        for(int i=0;i<9;i++){
            for(int j=0;j<9;j++){
                if(board[i][j]=='.'){
                    continue;
                }
                if(rows.get(i).contains(board[i][j]) || cols.get(j).contains(board[i][j]) || boxes.get((i/3)*3 + (j/3)).contains(board[i][j])){
                    return false;
                }
                rows.get(i).add(board[i][j]);
                cols.get(j).add(board[i][j]);
                boxes.get((i/3)*3+(j/3)).add(board[i][j]);
            }
        }
        return true;
    }
}